sdk/core/core-rest-pipeline/review/core-rest-pipeline.api.md

```diff
@@ -300,6 +303,35 @@ export interface RestErrorOptions {
     statusCode?: number;
 }
 
+// @public
+export class RetryError extends Error {
+    constructor(message?: string);
+    errors: RestError[];
+}
+
+// @public
+export function retryPolicy(...strategies: RetryStrategy[]): PipelinePolicy;
```

Comment:
@jeremymeng suggested removing the notion of retry strategies, but when I tried it the code looked way more complex. The separation of policies (now strategies) do help keep a separation of concerns. In some cases splitting logic into individual parts could be seen as a premature optimization, but since we already had the logic separated for those policies, I believe it would be a regression to merge them together.